rts/Lua/LuaConstCMD.cpp

```diff
@@ -60,11 +69,81 @@ bool LuaConstCMD::PushEntries(lua_State* L)
 
 #define PUSH_CMD(cmd) LuaInsertDualMapPair(L, #cmd, CMD_ ## cmd);
 
-	/*** @field CMD.STOP 0 */
+	/***
+	 * @field CMD.STOP 0
+	 *
+	 * Stop the current action.
+	 *
+	 * For factories, this will cancel the new unit orders queue.
+	 * For units, this will cancel the current command and queue.
+	 *
+	 * Accepts no parameters.
+	 *
+	 * *Note:* If `shift` option is used it will be a noop in the queue.
+	 */
 	PUSH_CMD(STOP);
 	/*** @field CMD.INSERT 1 */
 	PUSH_CMD(INSERT);
-	/*** @field CMD.REMOVE 2 */
+
+	/***
+	 * @field CMD.REMOVE 2
+	 *
+	 * Remove commands from a unit's queue.
+	 *
+	 * The behaviour can be modified with the following options:
+	 *
+	 * ## Modes of operation
+	 *
+	 * ### Filter by id or tag
+	 *
+	 * Removes all commands matching ids or tags from a provided array.
+	 *
+	 * - `params` {id1, id2 ...} or {tag1, tag2, ...} an array of ids or tags to look for, unless META is used too.
+	 * - With `alt`, remove by id, otherwise remove by tag.
+	 *
+	 * ### Remove by range
+	 *
+	 * Removes all commands in a given range, indexed by position in the queue, or command tag.
+	 *
+	 * - `params` {start, end}. both are optional and if not set default to queue limits.
+	 *   - Start and end will be clamped to the biggest queue intersection.
+	 *   - For example `{2,100}` is be processed as `{2,5}` (or `{2}`) in a queue with 5 items).
+	 * - With `alt`, start and end are indexes, without ALT, they're tags.
+	 *
+	 * ## Command Options
+	 * - `meta` Range/filter switch.
+	 * - `alt` Tag/Index or Tag/Id switch, depending on `meta`.
+	 * - `ctrl` Alternative queue selection.
+	 *   - For factories alternative queue is the factory command queue, default queue is the newUnitCommands queue.
```

Comment:
Avoid names of internal C++ objects
```suggestion
	 *   - For factories alternative queue is the factory command queue, default queue is the rally queue.
```